### PR TITLE
Disable HDCP on green_sardine and renoir

### PR DESCRIPTION
**Pull request by ivan-volnov**
Disable HDCP on green_sardine

A quick fix for #167

### CODE REVIEW

**Comment by ivan-volnov** [comment]
> Thanks this PR, I haved test it on my R7-5800U(green sardine), and it worked well.

I'm glad that it helped you, so it's really works.

**Comment by wulf7** [comment]
> Yes, but the origin drm module can work on 14-CURRENT, maybe we can try to find out which kernel change cause this error?
> 
> And my FreeBSD version is 13.1-RELEASE now.

Do you mean that `R7-5800U(green sardine)` works for you on CURRENT "as is" but fails on 13.1-RELEASE?

**Comment by asas1asas200** [comment]
> > Yes, but the origin drm module can work on 14-CURRENT, maybe we can try to find out which kernel change cause this error?
> > And my FreeBSD version is 13.1-RELEASE now.
> 
> Do you mean that `R7-5800U(green sardine)` works for you on CURRENT "as is" but fails on 13.1-RELEASE?

Yes. I test it about 2-3 weeks ago.

**Comment by evadot** [comment]
I'm not too fond of this patch.
I'd really like to know if the problem is on all green_sardine or only the ones for which we backported the pciids.

**Comment by ivan-volnov** [comment]
> It is better unconditionally ifdef-out whole HDCP firmware loading block. We do not support HDCP in drm-kmod yet

@wulf7 Ok, updated the PR.

@nkfilis Please test it on renoir. It should work now.

**Comment by nkfilis** [comment]
> @nkfilis Please test it on renoir. It should work now.

@ivan-volnov thank you, but unfortunately, the problem persists :(

**Comment by ivan-volnov** [comment]
@nkfilis Are you sure? It hasn't built until now. Fixed.

File: drivers/gpu/drm/amd/amdgpu/psp_v12_0.c [hunk 1]
```diff
@@ -49,9 +49,14 @@ static int psp_v12_0_init_microcode(struct psp_context *psp)
...
 	struct amdgpu_device *adev = psp->adev;
 	const char *chip_name;
+#ifdef __linux__
 	char fw_name[30];
 	int err = 0;
 	const struct ta_firmware_header_v1_0 *ta_hdr;
+#elif defined(__FreeBSD__)
+	/* We do not support HDCP in drm-kmod yet */
+	int err = 0;
+#endif
```
Comment (asas1asas200) [inline drivers/gpu/drm/amd/amdgpu/psp_v12_0.c:59]: Why not just do:
```c
#ifdef __linux__
    char fw_name[30];
    const struct ta_firmware_header_v1_0 *ta_hdr;
#endif
    /* We do not support HDCP in drm-kmod yet */
    int err = 0;
```

**Comment by nkfilis** [comment]
> @nkfilis Are you sure? It hasn't built until now. Fixed.

yes, I grabbed https://raw.githubusercontent.com/freebsd/drm-kmod/b3e6bb4d0a22e8c527e95afc4b0e261622b208b0/drivers/gpu/drm/amd/amdgpu/psp_v12_0.c after you asked, but still no dice. :(

Though it did build for me before after 982c604bc29ceb2bfab35852aac5aa5481985b69 as well.

File: drivers/gpu/drm/amd/amdgpu/psp_v12_0.c (hunk 1 above)
Comment (ivan-volnov) [reply drivers/gpu/drm/amd/amdgpu/psp_v12_0.c:59]: Tried to do as minimal changes as I could. Not to rearrange lines

**Comment by ivan-volnov** [comment]
@nkfilis Can you also try to merge the PR to the branch up to this 78361271cb8cfb758cccfdf911a0a19d3956ca9b commit? It works for me perfectly.

Also the error 'Cannot allocate memory' can be because of low GPU memory. Try to increase VRAM in BIOS.

**Comment by ivan-volnov** [comment]
> Yes, but the origin drm module can work on 14-CURRENT, maybe we can try to find out which kernel change cause this error?

> Yes. I test it about 2-3 weeks ago.

There was no kernel changes that caused the error. The driver was changed during these 2-3 weeks a lot. I also used old driver and it worked well for me. Now this PR fixes it on branch 5.10-lts.

@asas1asas200 so, this PR works for you too isn't it?

**Comment by nkfilis** [comment]
> Also the error 'Cannot allocate memory' can be because of low GPU memory. Try to increase VRAM in BIOS.

I can only increase it to 1024M and it worked with 512MB with 5.7.

**Comment by nkfilis** [comment]
> @nkfilis Can you also try to merge the PR to the branch up to this [7836127](https://github.com/freebsd/drm-kmod/commit/78361271cb8cfb758cccfdf911a0a19d3956ca9b) commit? It works for me perfectly.

It works! Thank you very much! Haven't tried setting VRAM back to 512MB, but it does work with 1024MB

**Comment by wulf7** [comment]
Please drop last `#elif defined(__FreeBSD__)` as it defines an empty block. Comment is not required here as it duplicates another one several lines before.

Also squash all your commits in single one and force push them after than

**Comment by ivan-volnov** [comment]
@wulf7 Done.

**Pull request merged**